tests/node_eviction_test.go

```diff
@@ -0,0 +1,289 @@
+	"kubevirt.io/kubevirt-ansible/tests"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+const (
+	cirrosImageDev  string = "registry:5000/kubevirt/cirros-registry-disk-demo:devel"
+	cirrosImageTest string = "kubevirt/cirros-registry-disk-demo:latest"
+)
+
+type vmi struct {
+	name     string
+	filename string
+	url      string
+}
+
+var _ = Describe("Node Eviction", func() {
+
+	flag.Parse()
+
+	var (
+		simpleVmi   vmi
+		vmBackedVmi vmi
+		rsBackedVmi vmi
+	)
+
+	BeforeEach(func() {
+		ktests.BeforeTestCleanup()
+
+		args := []string{"get", "node", "--selector=node-role.kubernetes.io/compute=true", "--output=jsonpath={.items..metadata.name}"}
+		output, err := ktests.RunOcCommand(args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		var count int
+		nodes := strings.Fields(output)
+		for _, node := range nodes {
+			args := []string{"get", "node", node}
+			output, err := ktests.RunOcCommand(args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			if !strings.Contains(output, "SchedulingDisabled") {
+				count += 1
+			}
+		}
+		if count < 2 {
+			panic(fmt.Errorf("no enough schedulble nodes for this testing"))
+		}
+
+		simpleVmi = vmi{
+			name:     "vmi-ephemeral",
+			filename: "/tmp/vmi-ephemeral.yaml",
+			url:      "https://raw.githubusercontent.com/kubevirt/kubevirt/master/cluster/examples/vmi-ephemeral.yaml",
+		}
+
+		vmBackedVmi = vmi{
+			name:     "vm-cirros",
+			filename: "/tmp/vm-cirros.yaml",
+			url:      "https://raw.githubusercontent.com/kubevirt/kubevirt/master/cluster/examples/vm-cirros.yaml",
+		}
+
+		rsBackedVmi = vmi{
+			name:     "vmi-replicaset-cirros",
+			filename: "/tmp/vmi-replicaset-cirros.yaml",
+			url:      "https://raw.githubusercontent.com/kubevirt/kubevirt/master/cluster/examples/vmi-replicaset-cirros.yaml",
+		}
+
+		tests.DownloadFromUrl(simpleVmi.url, simpleVmi.filename)
+		tests.DownloadFromUrl(vmBackedVmi.url, vmBackedVmi.filename)
+		tests.DownloadFromUrl(rsBackedVmi.url, rsBackedVmi.filename)
+		tests.ReplaceStringInFile(simpleVmi.filename, cirrosImageDev, cirrosImageTest, -1)
+		tests.ReplaceStringInFile(vmBackedVmi.filename, cirrosImageDev, cirrosImageTest, -1)
+		tests.ReplaceStringInFile(rsBackedVmi.filename, cirrosImageDev, cirrosImageTest, -1)
+	})
+
+	Describe("VirtualMachineInstance Evictions", func() {
+		It("virtualmachine instance will not be re-scheduled", func() {
+			By("Create the virtualmachine instance")
+			args := []string{"create", "-n", tests.NamespaceTestDefault, "-f", simpleVmi.filename}
+			_, err := ktests.RunOcCommand(args...)
+			Expect(err).ToNot(HaveOccurred())
+			time.Sleep(5)
+
+			By("Check the virtualmachine instance is running")
+			for {
+				args := []string{"get", "-n", tests.NamespaceTestDefault, "vmi", simpleVmi.name, "--template", "{{.status.phase}}"}
+				output, err := ktests.RunOcCommand(args...)
+				Expect(err).ToNot(HaveOccurred())
+				if output == "Running" {
+					break
+				}
+				// have a time sleep to make the query not so frequently
```

Comment:
This sentence is not correct. I would write:
`wait so that the query does not run so frequently`